src/RestControllers/AuthorizationController.php

```diff
@@ -1005,4 +1005,86 @@ public function tokenIntrospection(): void
         $this->emitResponse($response->withStatus(200)->withBody($body));
         exit();
     }
+
+    /**
+     * Returns the authentication server token Url endpoint
+     * @return string
+     */
+    public function getTokenUrl() {
+        return $this->authBaseUrl . self::getTokenPath();
+    }
+
+    /**
+     * Returns the path prefix that the token authorization endpoint is on.
+     * @return string
+     */
+    public static function getTokenPath() {
+        return "/token";
+    }
+
```

Comment:
I'm just wondering if the better option instead of hard coding auth server endpoints is to pull in the endpoints from auth server discovery.
This way we can pull in all the servers supported claims/scopes/endpoints where, as they change, we have a single point of reference.

Unsure SMART needs these references in resource server however, while here, worthwhile considering.

Reply:
So I like keeping the endpoint URLs centralized.   However, the FhirMetadataRestController needs the endpoint information as does the SmartConfigurationController.  Since the FHIR baseURL is controlled by AuthorizationController I thought it best to put all of that path information into the AuthorizationController.

I didn't modify discover.php, but I would actually change discover.php's URLs to use the AuthorizationController.
I need to take a glance and make sure the Resource Read/Write scopes in the smart well-known configuration are the same in as what we show in the discover.php file.  If they are we can centralize them also in AuthorizationController.  If you don't like it in the AuthorizationController than we can centralize it in discover.php but that file outputs JSON and header requests and it feels a bit odd trying to bring those into the other places it needs.  I'd rather keep it in a class like AuthorizationController or a different class that can be reused everywhere else.

Reply:
Right O! lol, I thought I reviewing the rest_config singleton! Told ya, i'm horrible at reviewing code! This is in the appropriate place...